tests/sim/utils.rs

```diff
@@ -20,6 +22,33 @@ pub fn assert_eq_with_gas(left: u128, right: u128) {
     assert_almost_eq_with_max_delta(left, right, to_yocto("0.005"));
 }
 
+/// Util to check a balance is the same as in a previous state
+pub fn assert_unchanged_balance(account: UserAccount, previous_balance: u128, ft_contract: &ContractAccount<FungibleTokenContractContract>, account_name: &str) {
+    
+    let current_balance = call!(account, ft_contract.ft_balance_of(account.account_id()))
+        .unwrap_json::<U128>()
+        .0;
+    assert!(current_balance == previous_balance, "{}'s balance changed by {} (from {} to {})", account_name, previous_balance-current_balance, previous_balance, current_balance);
+}
+
+/// Util to assert that an account has spent a given amount of token.
+pub fn assert_spent(account: UserAccount, previous_balance: u128, expected_spent_amount: u128, ft_contract: &ContractAccount<FungibleTokenContractContract>) {
+    let current_balance = call!(account, ft_contract.ft_balance_of(account.account_id()))
+        .unwrap_json::<U128>()
+        .0;
+    assert!(current_balance <= previous_balance, "Did not spend.");
+    assert!(current_balance == previous_balance -expected_spent_amount, "Spent {} instead of {}", previous_balance - current_balance, expected_spent_amount);
```

Comment:
Please add spaces around operators.


```suggestion
    assert!(current_balance == previous_balance - expected_spent_amount, "Spent {} instead of {}", previous_balance - current_balance, expected_spent_amount);
```